src/main/java/converters/IntConverter.java

```diff
@@ -0,0 +1,178 @@
+import static options.IntOption.LITTLE;
+
+/**
+ * This class provides methods for converting between unsigned integer value strings and binary strings.
+ * It extends the abstract {@link Converter} class.
+ * <p>
+ * To convert an unsigned integer string to a binary string, use the {@link #convertTo(String, IOption[])} method.
+ * To convert a binary string to an unsigned integer value string, use the
+ * {@link #convertFrom(String, IOption[])} method.
+ * <p>
+ * The IntConverter supports the following options:
+ * <ul>
+ *     <li>{@link IntOption#BIG}: big-endian format (default)</li>
+ *     <li>{@link IntOption#LITTLE}: little-endian format</li>
+ * </ul>
+ */
+public class IntConverter extends Converter {
+
+	private static final int TEN = 10;
+	/**
+	 * Converts a binary string to an unsigned integer value string.
+	 *
+	 * @param bitStr   the binary string to convert
+	 * @param options  an array of {@link IntOption} options (first is taken,
+	 *                    if none is provided, {@link IntOption#BIG} is used)
+	 * @return the unsigned integer value string
+	 * @throws ConverterException if the input binary string is invalid
+	 */
+	@Override
+	public String convertTo(String bitStr, IOption[] options) throws ConverterException {
+		validateInput(bitStr, "^[01]+$");
+		IntOption endian = getEndianFromOptions(options);
+
+		String updatedBitStr = processBitString(bitStr, endian);
+
+		String decimalNumber = STRING_WITH_ZERO;
+		String powerOfTwo = "1";
+		for (int i = updatedBitStr.length() - 1; i >= 0; i--) {
+			int digit = updatedBitStr.charAt(i) - '0';
+			if (digit == 0) {
+				powerOfTwo = multiplyStrings(powerOfTwo, "2");
+				continue;
+			}
+
+			String product = multiplyStrings(String.valueOf(digit), powerOfTwo);
+			decimalNumber = addStrings(decimalNumber, product);
+			powerOfTwo = multiplyStrings(powerOfTwo, "2");
+		}
+		return decimalNumber;
+	}
+
+	/**
+	 * Converts an unsigned integer value string to a binary string.
+	 *
+	 * @param input    the unsigned integer value string to convert
+	 * @param options  an array of {@link IntOption} options (first is taken,
+	 *                    if none is provided, {@link IntOption#BIG} is used)
+	 * @return the binary string
+	 * @throws ConverterException if the input unsigned integer value string is invalid
+	 */
+	@Override
+	public String convertFrom(String input, IOption[] options) throws ConverterException {
+		validateInput(input, "^\\d+$");
+		IntOption endian = getEndianFromOptions(options);
+
+		StringBuilder binary = new StringBuilder();
+		StringBuilder updatedInput = new StringBuilder(input);
+
+		while (updatedInput.length() > 0) {
+			StringBuilder semiResult = new StringBuilder();
+			boolean foundNoneZero = false;
+			int remainder = 0;
+
+			for (int i = 0; i < updatedInput.length(); i++) {
+				int digit = Character.getNumericValue(updatedInput.charAt(i));
+				int dividend = remainder * TEN + digit;
+				char charDigit = (char) ('0' + (dividend / 2));
```

Comment:
thought: Why not make a constant named TWO, since there is a constant named TEN?